src/modm/platform/gpio/sam/pin.hpp

```diff
@@ -28,24 +28,6 @@ enum class InputType
 	PullDown = 0x2,  ///< pull-down on input
 };
 
-enum class OutputType
-{
-	PushPull = 0x0,   ///< push-pull on output
-	OpenDrain = 0x1,  ///< open-drain on output
-};
```

Comment:
Hm interesting, SAM devices seem to have no OpenDrain config? In that case you will need [this helper class from AVRs](https://github.com/modm-io/modm/blob/develop/src/modm/platform/gpio/at90_tiny_mega/base.hpp.in#L88-L144) and also [enable the use of it in the software I2C](https://github.com/modm-io/modm/blob/develop/src/modm/platform/i2c/bitbang/bitbang_i2c_master.hpp.in#L41-L47).

Reply:
The SAMG seems not to, for software GPIO control -- which seems sensible to me, because you get to choose between tri-state, high, low: why do you need a mode to convert high to tri for you? But it does have a mode to make peripheral controlled IOs open drain. I still need to revisit this, it may just be something that has to get configured by the user, or maybe it can be part of the `connect` process for peripherals like I2C that need it that way. 

I actually haven't looked at any of the SAMD datasheets, but the current SAM GPIO driver didn't do anything with the OpenDrain setting, so I didn't feel too bad about axing it.